Description: Add cpu feature pks.
 Add the pks feature definition along with its entries in QEMU feature lists,
 ignoring the 7.* entries from this commit upstream.
Author: Tim Wiederhake <twiederh@redhat.com>
Origin: backport, https://gitlab.com/libvirt/libvirt/-/commit/40d6a6d98702452df7c034bfe81e3a334f0f8e45
Bug-Ubuntu: https://launchpad.net/bugs/2028057
Forwarded: not-needed
Last-Update: 2023-11-29
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/cpu_map/x86_features.xml
+++ b/src/cpu_map/x86_features.xml
@@ -349,6 +349,9 @@
   <feature name='sgxlc'>
     <cpuid eax_in='0x07' ecx_in='0x00' ecx='0x40000000'/>
   </feature>
+  <feature name='pks'>
+    <cpuid eax_in='0x07' ecx_in='0x00' ecx='0x80000000'/>
+  </feature>
 
   <feature name='avx512-4vnniw'>
     <cpuid eax_in='0x07' ecx_in='0x00' edx='0x00000004'/>
--- a/tests/domaincapsdata/qemu_6.0.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.0.0-tcg.x86_64.xml
@@ -48,6 +48,7 @@
       <feature policy='require' name='clwb'/>
       <feature policy='require' name='pku'/>
       <feature policy='require' name='la57'/>
+      <feature policy='require' name='pks'/>
       <feature policy='require' name='3dnowext'/>
       <feature policy='require' name='3dnow'/>
       <feature policy='require' name='npt'/>
--- a/tests/domaincapsdata/qemu_6.1.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.1.0-tcg.x86_64.xml
@@ -48,6 +48,7 @@
       <feature policy='require' name='clwb'/>
       <feature policy='require' name='pku'/>
       <feature policy='require' name='la57'/>
+      <feature policy='require' name='pks'/>
       <feature policy='require' name='3dnowext'/>
       <feature policy='require' name='3dnow'/>
       <feature policy='require' name='npt'/>
--- a/tests/domaincapsdata/qemu_6.2.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.2.0-tcg.x86_64.xml
@@ -48,6 +48,7 @@
       <feature policy='require' name='clwb'/>
       <feature policy='require' name='pku'/>
       <feature policy='require' name='la57'/>
+      <feature policy='require' name='pks'/>
       <feature policy='require' name='3dnowext'/>
       <feature policy='require' name='3dnow'/>
       <feature policy='require' name='npt'/>
